tools/rosbag_storage/src/bag.cpp

```diff
@@ -217,6 +218,14 @@ void Bag::setCompression(CompressionType compression) {
     compression_ = compression;
 }
 
+void Bag::setEncryptorPlugin(std::string const& plugin_name, std::string const& plugin_param) {
```

Comment:
How much overlap is there between the encryption functions and the compression functions? My question about this would be whether it would make more sense to have this as a more generic `setStoragePlugin`, where bzip2 and lzma compression plugins can be loaded similar to the encryption scheme.

Obviously I don't want the scope to get away from us, but it's just a thought.

Reply:
Would `setStoragePlugin` cover (i) encryption, (ii) compression, and (iii) the combination of the two?

Reply:
Honestly, I haven't given it that much thought. If there isn't an obvious path on that, disregard— a plugin hook which applies only to encryption is fine with me.

Reply:
Thank you for your input @mikepurvis.  Nothing is wrong with having a single 'storage' plugin (except that we might end up with `m`x`n` plugin classes for `m` compression methods and `n` encryptors).  Refactoring existing compression stream classes and combining them with encryptors would require lots of efforts, and I would like leave that as a future task.